### PR TITLE
Admin add movie

### PR DESCRIPTION
**Pull request by nhat15**
Admin add show

## Related Tickets
- [#52717](https://edu-redmine.sun-asterisk.vn/issues/52717)

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
Screenshot from 2022-08-23 14-50-30
Screenshot from 2022-08-23 14-50-56

### CODE REVIEW

**Comment by nhat15** [comment]
ready

File: app/models/show.rb [hunk 1]
```diff
@@ -2,7 +2,25 @@ class Show < ApplicationRecord
...
+  validate :check_duration_time
 
   scope :asc_date, ->{order date: :asc}
   scope :asc_time, ->{order start_time: :asc}
+
+  delegate :title, to: :movie, prefix: :movie
+  delegate :name, to: :room, prefix: :room
+
+  def check_duration_time
+    return if Show.where("date = ? AND (start_time < ? OR end_time > ?)
```
Comment (huongnt-2545) [inline app/models/show.rb]:
```
Show.where("date = ? AND (start_time < ? OR end_time > ?)
    AND room_id = ?", date, end_time, start_time, room_id)
```
-> đặt biến dùng đi em

File: app/models/show.rb [hunk 2]
```diff
@@ -2,7 +2,25 @@ class Show < ApplicationRecord
...
+    AND room_id = ?", date, end_time, start_time, room_id).none?
+
+    errors.add(:start_time, "Times overlap another showtime on this screen")
```
Comment (huongnt-2545) [inline app/models/show.rb]: I18n text

File: app/views/admin/layouts/_sidebar.html.erb [hunk 3]
```diff
@@ -1,38 +1,38 @@
...
-          <a href="#" class="nav_link">
-            <i class='fa fa-credit-card nav_icon'></i>
-            <span class="nav_name"><%= t ".revenue" %></span>
-          </a>
-        </div>
-      </div>
-      <a href="#" class="nav_link"> <i class='fa fa-sign-out'></i> <span class="nav_name"><% t ".signout" %></span> </a>
-    </nav>
-  </div>
+    <header class="header" id="header">
```
Comment (huongnt-2545) [inline app/views/admin/layouts/_sidebar.html.erb]: check lại indent file này

**Comment by nhat15** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**